File: capacity-club-api/src/domain-modules/member-plan/model/entity/member-plan.entity.ts
import { Column, Entity, PrimaryColumn } from 'typeorm';
import { BaseEntity } from '@common/model';
import { UniqueId, uniqueIdTransformer } from '@common/model/unique-id';
import { Transform } from 'class-transformer';
/**
 * Entity representing member plans.
 * These plans are used to recharge training credits on a member's card.
 */
@Entity()
export class MemberPlan extends BaseEntity {
  /**
   * Unique identifier for the member plan.
   * Uses a custom transformer for serialization.
   */
  @PrimaryColumn('varchar')
  @Transform(uniqueIdTransformer.to, { toClassOnly: true })
  @Transform(uniqueIdTransformer.from, { toPlainOnly: true })
  member_plan_id: UniqueId;
  @Column({ length: 80, nullable: false })
  title: string;
  @Column('text', { nullable: true })
  description: string;
  @Column({ default: 10 })
  nb_individual_training: number;
  @Column({ default: 10 })
  nb_collective_training: number;
  @Column({ nullable: false })
  price: number;
}

File: capacity_club_mobile_app/lib/domain-features/domain/entities/member_plan_entity.dart
import 'package:capacity_club_mobile_app/core/model/entities/unique_id.dart';

class MemberPlanEntity {
  final UniqueId member_plan_id;
  final String title;
  final String description;
  final int nb_individual_training;
  final int nb_collective_training;
  final double price;
  MemberPlanEntity(
      {required this.member_plan_id,
      required this.title,
      required this.description,
      required this.nb_individual_training,
      required this.nb_collective_training,
      required this.price});

  factory MemberPlanEntity.empty() {
    return MemberPlanEntity(
        member_plan_id: UniqueId(''),
        title: '',
        description: '',
        nb_individual_training: 0,
        nb_collective_training: 0,
        price: 0);
  }
}
